src/Handlers/ShortcodeParametersTransformHandler.php

```diff
@@ -0,0 +1,191 @@
+     *
+     * @var int
+     */
+    protected $cartPageId;
+
+    /**
+     * Resource model for selecting services.
+     *
+     * @since [*next-version*]
+     *
+     * @var SelectCapableInterface
+     */
+    private $serviceSelectResourceModel;
+
+    /**
+     * Expression builder.
+     *
+     * @since [*next-version*]
+     *
+     * @var ExpressionBuilderInterface
+     */
+    private $expressionBuilder;
+
+    /**
+     * Service transformer.
+     *
+     * @since [*next-version*]
+     *
+     * @var TransformerInterface
+     */
+    private $serviceTransformer;
+
+    /**
+     * MainComponentHandler constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param int|string|Stringable|float $cartPageId                 Cart page ID.
+     * @param SelectCapableInterface      $serviceSelectResourceModel Resource model for selecting services.
+     * @param ExpressionBuilderInterface  $expressionBuilder          Expression builder.
+     * @param TransformerInterface        $serviceTransformer         Service transformer.
+     */
+    public function __construct($cartPageId, $serviceSelectResourceModel, $expressionBuilder, $serviceTransformer)
+    {
+        $this->cartPageId                 = $this->_normalizeInt($cartPageId);
+        $this->serviceSelectResourceModel = $serviceSelectResourceModel;
+        $this->expressionBuilder          = $expressionBuilder;
+        $this->serviceTransformer         = $serviceTransformer;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        /* @var $event EventInterface */
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        $event->setParams($this->_handleParameters($event->getParams()));
+    }
+
+    /**
+     * Render booking holder.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $params List of shortcode parameters.
+     *
+     * @return array Prepared parameters for wizard.
+     */
+    protected function _handleParameters($params = [])
+    {
+        $params['redirectUrl'] = $this->_getRedirectUrl();
+
+        if (isset($params['service'])) {
+            $service = $this->_getService((int) $params['service']);
+            unset($params['service']);
+            if ($service) {
+                $params['service'] = $this->_normalizeArray($service);
+            }
+        }
+
+        return $params;
+    }
+
+    /**
+     * Get cart URL on which customer will be redirected after successful booking creation.
+     *
+     * @since [*next-version*]
+     *
+     * @return string Cart URL to redirect user on.
+     */
+    protected function _getRedirectUrl()
+    {
+        $pageUrl = get_permalink($this->cartPageId);
+
+        if (!$pageUrl) {
```

Comment:
The value that indicates error is `false`, not everything that evaluates to `false`.